recipe/build.sh

```diff
@@ -51,6 +51,7 @@ sed -i.bak "s,cc,${TOOLSET},g" ${SRC_DIR}/project-config.jam
     toolset=${TOOLSET}-custom \
     include="${INCLUDE_PATH}" \
     cxxflags="${CXXFLAGS}" \
+    cxxstd=14 \
```

Comment:
As this isn't an option with the legacy compilers on Linux, we could add a branch that checks the environment variable `cxx_compiler` to see that it is not `"toolchain_cxx"` before setting. That should fix the CI failure.